velox/functions/sparksql/aggregates/SumAggregate.cpp

```diff
@@ -24,9 +25,30 @@ namespace facebook::velox::functions::aggregate::sparksql {
 namespace {
 template <typename TInput, typename TAccumulator, typename ResultType>
 using SumAggregate = SumAggregateBase<TInput, TAccumulator, ResultType, true>;
+
+TypePtr getDecimalSumType(
+    const TypePtr& resultType,
+    core::AggregationNode::Step step) {
+  if (exec::isPartialOutput(step)) {
+    return resultType->childAt(0);
+  }
+  if (step == core::AggregationNode::Step::kSingle && resultType->isRow()) {
```

Comment:
@liujiayi771 What would it be the case that step == kSingle and resultType is a ROW?

Reply:
For companion function. This section was added by @rui-mo, maybe she can provide more details.

Reply:
Hi @mbasmanova, this is for companion function of decimal sum. For a partial companion function of single step, the result type is like partial output, consisting of sum and isEmpty.

Reply:
@rui-mo Got it. Thank you for clarifying. It would be nice to add comment in the code for the benefit of future readers.